src/containers/VerletListHelpers.h

```diff
@@ -45,7 +45,102 @@ class VerletListHelpers {
         // (ensured by traversals)
         // also the list is not allowed to be resized!
 
-        _verletListsAoS[&i].push_back(&j);
+        _verletListsAoS.at(&i).push_back(&j);
+    }
+
+    virtual void SoAFunctor(SoA &soa, bool /*newton3*/ = true) override {
+      if (soa.getNumParticles() == 0) return;
+
+      Particle **const __restrict__ idptr = reinterpret_cast<Particle **const>(soa.begin(Particle::AttributeNames::id));
+      double *const __restrict__ xptr = soa.begin(Particle::AttributeNames::posX);
+      double *const __restrict__ yptr = soa.begin(Particle::AttributeNames::posY);
+      double *const __restrict__ zptr = soa.begin(Particle::AttributeNames::posZ);
+
+      size_t numPart = soa.getNumParticles();
+      for (unsigned int i = 0; i < numPart; ++i) {
+        auto &currentList = _verletListsAoS.at(idptr[i]);
+
+        for (unsigned int j = i + 1; j < numPart; ++j) {
```

Comment:
we could check if it is worthwhile to vectorize this with `omp simd`

Reply:
see below